administrator/components/com_joomgallery/models/maintenance.php

```diff
@@ -499,13 +499,22 @@ public function delete($refids = true)
     // Loop through selected images
     foreach($cids as $cid)
     {
-      $error = false;
-
       if(!$row->load($cid))
       {
         continue;
       }
 
+      // Check whether we are allowed to delete this image
+      if(   !$this->_user->authorise('core.delete', _JOOM_OPTION.'.image.'.$cid)
+        && (!$this->_user->authorise('joom.delete.own', _JOOM_OPTION.'.image.'.$cid) || !$row->owner || $row->owner != $this->_user->get('id')))
```

Comment:
'joom.delete.own' is not defined in your branch and so it is not possible to test this code. You should rebase your branch to the current v3.6.0 branch as only this branch contains the changes regarding the 'joom.delete.own' permission.

Reply:
should be available now...